### PR TITLE
feat(jax/array-api): DPA-2

### PR DESCRIPTION
**Pull request by njzjz**
## Summary by CodeRabbit

- **New Features**
  - Introduced new classes for enhanced descriptor functionality, including `DescrptDPA2`, `DescrptBlockRepformers`, and `DescrptBlockSeTTebd`.
  - Added serialization and deserialization methods for better state management of descriptor objects.

- **Improvements**
  - Enhanced compatibility with various array backends through the integration of `array_api_compat`.
  - Refactored existing methods to utilize new array API functions for improved performance.
  - Updated documentation to reflect JAX as a supported backend alongside PyTorch.

- **Bug Fixes**
  - Updated handling of attributes in several classes to ensure correct deserialization and type safety.

- **Tests**
  - Enhanced testing capabilities for JAX and Array API Strict backend integration, including conditional imports and new evaluation methods.

### CODE REVIEW

File: deepmd/jax/descriptor/dpa2.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        elif name in {"repinit"}:
+            value = DescrptBlockSeAtten.deserialize(value.serialize())
+        elif name in {"repinit_three_body"}:
+            if value is not None:
+                value = DescrptBlockSeTTebd.deserialize(value.serialize())
+        elif name in {"repformers"}:
+            value = DescrptBlockRepformers.deserialize(value.serialize())
+        elif name in {"type_embedding"}:
+            value = TypeEmbedNet.deserialize(value.serialize())
+        elif name in {"g1_shape_tranform", "tebd_transform"}:
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/jax/descriptor/dpa2.py:51]: @iProzd I found all `g1_shape_transform` in the code are `g1_shape_tranform`...

File: deepmd/dpmodel/descriptor/se_t_tebd.py [hunk 2]
```diff
@@ -652,82 +663,85 @@
...
-            np.expand_dims(atype_embd_nlist, axis=1), [1, self.nnei, 1, 1]
-        )
+        nlist_tebd_i = xp.tile(atype_embd_nlist[:, :, None, :], (1, 1, self.nnei, 1))
+        nlist_tebd_j = xp.tile(atype_embd_nlist[:, None, :, :], (1, self.nnei, 1, 1))
         ng = self.neuron[-1]
 
         if self.tebd_input_mode in ["concat"]:
             # nfnl x nt_i x nt_j x (1 + tebd_dim * 2)
-            ss = np.concatenate([ss, nlist_tebd_i, nlist_tebd_j], axis=-1)
+            ss = xp.concat([ss, nlist_tebd_i, nlist_tebd_j], axis=-1)
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/dpmodel/descriptor/se_t_tebd.py:716]: Please note that only `xp.concat` is available.

File: deepmd/dpmodel/descriptor/se_t_tebd.py
Comment (njzjz) [inline deepmd/dpmodel/descriptor/se_t_tebd.py:1]: This file is just copied from #4286. May merge #4286 first.

File: source/tests/array_api_strict/descriptor/se_t_tebd.py
Comment (njzjz) [inline source/tests/array_api_strict/descriptor/se_t_tebd.py:1]: This file is just copied from #4286. May merge #4286 first.

**Review by wanghan-iapcm — APPROVED** [review]

File: deepmd/dpmodel/descriptor/se_t_tebd.py
Comment (njzjz) [reply deepmd/dpmodel/descriptor/se_t_tebd.py:1]: Sorry I mean #4288. Not #4286.

File: source/tests/array_api_strict/descriptor/se_t_tebd.py
Comment (njzjz) [reply source/tests/array_api_strict/descriptor/se_t_tebd.py:1]: Sorry I mean #4288. Not #4286.

File: deepmd/jax/descriptor/dpa2.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/jax/descriptor/dpa2.py:51]: We decide to keep it, just for compatibility.

**Pull request merged**